labelbox/data/annotation_types/data/tiled_image.py

```diff
@@ -0,0 +1,364 @@
+        leaflet reports all points as pixel locations at a zoom of 0
+        """
+        xend, xstart, yend, ystart = (
+            self.tile_bounds.bounds[1].x,
+            self.tile_bounds.bounds[0].x,
+            self.tile_bounds.bounds[1].y,
+            self.tile_bounds.bounds[0].y,
+        )
+        return (*[
+            x * (2**(zoom)) / self.tile_size
+            for x in [xstart, ystart, xend, yend]
+        ],)
+
+    def _get_3857_image_params(self, zoom) -> Tuple[float, float, float, float]:
+        """Computes the x and y tile bounds for fetching an image that
+        captures the entire labeling region (TiledData.bounds) given a specific zoom
+        """
+        lat_start, lat_end = self.tile_bounds.bounds[
+            1].y, self.tile_bounds.bounds[0].y
+        lng_start, lng_end = self.tile_bounds.bounds[
+            1].x, self.tile_bounds.bounds[0].x
+
+        # Convert to zoom 0 tile coordinates
+        xstart, ystart = self._latlng_to_tile(lat_start, lng_start, zoom)
+        xend, yend = self._latlng_to_tile(lat_end, lng_end, zoom)
+
+        # Make sure that the tiles are increasing in order
+        xstart, xend = min(xstart, xend), max(xstart, xend)
+        ystart, yend = min(ystart, yend), max(ystart, yend)
+        return (*[pt * 2.0**zoom for pt in [xstart, ystart, xend, yend]],)
+
+    def _latlng_to_tile(self,
+                        lat: float,
+                        lng: float,
+                        zoom=0) -> Tuple[float, float]:
+        """Converts lat/lng to 3857 tile coordinates
+        Formula found here:
+        https://wiki.openstreetmap.org/wiki/Slippy_map_tilenames#lon.2Flat_to_tile_numbers_2
+        """
+        scale = 2**zoom
+        lat_rad = math.radians(lat)
+        x = (lng + 180.0) / 360.0 * scale
+        y = (1.0 - math.asinh(math.tan(lat_rad)) / math.pi) / 2.0 * scale
+        return x, y
+
+    def _tile_to_pixel(self, tile: float) -> Tuple[int, int]:
+        """Rounds a tile coordinate and reports the remainder in pixels
+        """
+        rounded_tile = int(tile)
+        remainder = tile - rounded_tile
+        pixel_offset = int(self.tile_size * remainder)
+        return rounded_tile, pixel_offset
+
+    def _fetch_image_for_bounds(
+        self,
+        x_tile_start: int,
+        y_tile_start: int,
+        x_tile_end: int,
+        y_tile_end: int,
+        zoom: int,
+    ) -> np.ndarray:
+        """Fetches the tiles and combines them into a single image
+        """
+        tiles = {}
+        with ThreadPoolExecutor(max_workers=TILE_DOWNLOAD_CONCURRENCY) as exc:
+            for x in range(x_tile_start, x_tile_end + 1):
+                for y in range(y_tile_start, y_tile_end + 1):
+                    tiles[(x, y)] = exc.submit(self._fetch_tile, x, y, zoom)
+
+        rows = []
+        for y in range(y_tile_start, y_tile_end + 1):
+            rows.append(
+                np.hstack([
+                    tiles[(x, y)].result()
```

Comment:
Use np.pad based on your findings that the tiles might not be guarenteed to be TILE_SIZE. You can pad with zeros but maybe make that a param of the class too.